src/org/etools/j1939_84/controllers/part01/Part01Step26Controller.java

```diff
@@ -300,47 +310,588 @@ protected void run() throws Throwable {
+                                                                           .peek(this::save)
+                                                                           .collect(Collectors.toList());
+        if (!ghg100HrPackets.isEmpty()) {
+            // 6.1.26.17.c. List data received in a table using lifetime, stored 100 hr,
+            // active 100hr for columns, and categories for rows.
+            getListener().onResult(ghgTrackingModule.formatTechTable(Stream.concat(ghgPackets.stream(),
+                                                                                   ghg100HrPackets.stream())
+                                                                           .collect(Collectors.toList())));
+        }
+
+        for (int pg : List.of(GHG_ACTIVE_GREEN_HOUSE_100_HR, GHG_STORED_GREEN_HOUSE_100_HR)) {
+            GenericPacket ghgTracking100HrPacket = haveResponseWithPg(ghg100HrPackets, pg);
+            if (ghgTracking100HrPacket == null) {
+                if (getEngineModelYear() >= 2024) {
+                    // FIXME: current version of the document says warning; Eric clarified to be a failure
+                    // @Joe - we just need to remove above comment when Eric gives us a document with it corrected.
+                    // 6.1.26.18.a. For all MY2024+ engines, Warn each PG query where no response was received.
+                    addFailure("6.1.26.18.a - No response was received from "
+                            + module.getModuleName() + " for PG "
+                            + pg);
+                }
+                if (getEngineModelYear() >= 2022 && getEngineModelYear() <= 2023) {
+                    // 6.1.26.18.b. For MY2022-23 engines, Warn each PG query, where no response was received
+                    addWarning("6.1.26.18.b - No response was received from "
+                            + module.getModuleName() + " for PG "
+                            + pg);
+                }
+            } else {
+                ghgTracking100HrPacket.getSpns()
+                                      .forEach(spn -> {
+                                          // 6.1.26.18.c. Fail PG query where any bin value received is greater than
+                                          // FAFFh.
+                                          if (spn.getRawValue() >= 0xFAFFL) {
+                                              addFailure("6.1.26.18.c - Bin value received is greater than 0xFAFF(h)"
+                                                      + module.getModuleName() + " for " + spn);
+                                          }
+                                          // 6.1.26.18.d. Fail PG query where any index value received is greater than
+                                          // FAh.
+                                          if (spn.getId() == 12691 && spn.getValue() > 0xFAL) {
+                                              addFailure("6.1.26.16.c - PG query index received was " + "");
```

Comment:
I think this should be "6.1.26.16.d"

Reply:
Also the message is odd.  Shouldn't we include the module name that produced the error?